python/lsst/ts/envsensors/command_handler.py

```diff
@@ -266,8 +284,13 @@ async def connect_devices(self) -> None:
         device_configurations = self._configuration[Key.DEVICES]  # type: ignore
         self._devices = []
         for device_configuration in device_configurations:
-            device = self._get_device(device_configuration)
+            device: BaseDevice = self._get_device(device_configuration)
             self._devices.append(device)
+            self.log.info(
+                f"Opening {device_configuration[Key.DEVICE_TYPE]} "
+                f"device with name {device_configuration[Key.NAME]}"
+            )
+            await device.open()
```

Comment:
I worry that the new API makes it easy to screw up starting and stopping by getting the two operations in the wrong order.

Please provide a way to do these two operations in a single call. I suggest that you make start call open and stop call close. If you want to be able to avoid the extra call you could add a flag (which I suggest be required, in order to avoid ambiguity), e.g. `def start(self, open):...`

A simpler approach is to only check for the minimal keys here. Then feed sensor configuration (pop the few non-device-config keys first) as a keywords argument dict to the appropriate sensor constructor. The constructor will raise an exception if values are missing or unrecognized values are present. This avoids duplicating information and makes the sensor constructor the source of truth. It may lead to slightly less clear error messages, but you can do a lot by using try/catch to catch and supplement the exception from the sensor constructor.

Reply:
Agreed. I merged the `open` and `start` functions and also the `stop` and `close` functions. If a device gets opened, the telemetry loop needs to start and if a device gets closed, the telemetry loop needs to stop.